contracts/erc20-gateway/Gen0ERC20Cogateway.sol

```diff
@@ -0,0 +1,86 @@
+pragma solidity >=0.5.0 <0.6.0;
+
+// Copyright 2020 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import "./ERC20Cogateway.sol";
+
+/**
+ * @title Implements ERC20Cogateway contract for gen0.
+ */
+contract Gen0ERC20Cogateway is ERC20Cogateway {
+
+    /* External Functions */
+
+    /**
+     * @notice activate() function activates the contract.
+     *
+     * @dev The function sets corresponding genesis* variables inherited
+     *      from GenesisERC20Cogateway contract and calls the parent class
+     *      ERC20Cogateway::setup() function to initialize contract.
+     *
+     * \pre `_metachainId` is not 0.
+     * \pre `_erc20Gateway` is not 0.
+     * \pre `_stateRootProvider` is not 0.
+     * \pre `maxStorageRootItems` is not 0.
+     *
+     * \post Sets corresponding genesis* variables inherited from
```

Comment:
Should there be a separate line for each \post. 
e.g. 
\post Sets the `genesisMetachainId` storage variable with `_metachainId`.
\post Sets the `genesisERC20Gateway` storage variable with `_erc20Gateway`.
..
..
..
\post Calls ERC20Cogateway::setup()

Reply:
Thinking was that the list of genesis* variables are defined in the genesis contract, and function arguments are exactly matching those variables, that's why for simplicity without breaking readability and easiness it has been done so. Do you have a feeling that it's not clear from the current `\post` function what is happening?

Reply:
It can be split like below which I think is more clear:
```
\post Sets genesisMetachainId, genesisERC20Gateway, genesisStateRootProvider, genesisMaxStorageRootItems, genesisOutboxStorageIndex storage variables.
\post Calls ERC20Cogateway::setup() function to initialize the contract. Calls to ERC20Cogateway::setup() calls MessageOutbox::setupMessageOutbox() which assures that activate() function cannot be called only once.
```

Reply:
I respectfully disagree here. Having one post condition here is clear to me, as the only reason of setting genesis* variables are being able to call setup() function. However, I will only update it as proposed to move ticket forward.